test/unit/subnet/ipv6_test.rb

```diff
@@ -0,0 +1,149 @@
+require 'test_helper'
+
+class SubnetIpv6Test < ActiveSupport::TestCase
+  def setup
+    User.current = User.find_by_login "admin"
+  end
+
+  test "should have a network" do
+    @subnet = FactoryGirl.build(:subnet_ipv6)
+    old_network = @subnet.network
+    @subnet.network = nil
+    refute @subnet.valid?
+
+    @subnet.network = old_network
+    assert @subnet.valid?
+  end
+
+  test "should have a mask" do
+    @subnet = FactoryGirl.build(:subnet_ipv6)
+    old_mask = @subnet.mask
+    @subnet.mask = nil
+    refute @subnet.valid?
+
+    @subnet.mask = old_mask
+    assert @subnet.valid?
+  end
+
+  test "cidr setter should set the mask" do
+    @subnet = FactoryGirl.build(:subnet_ipv6)
+    @subnet.cidr = 64
+    assert_equal 'ffff:ffff:ffff:ffff::', @subnet.mask
+  end
+
+  test "network should have ip format" do
+    @subnet = FactoryGirl.build(:subnet_ipv6, :network => "asf:fwe6::we6s:q1")
+    refute @subnet.valid?
+  end
+
+  test "mask should have ip format" do
+    @subnet = FactoryGirl.build(:subnet_ipv6, :mask => "asf:fwe6::we6s:q1")
+    refute @subnet.valid?
+  end
+
+  test "network should be unique" do
```

Comment:
Actually that comes from the parent class. so you should modify `test/unit/subnet.rb` to include these validations that are not specific from ipv6 to put them there. I saw `ipv4_test` also contains a bunch of these, just test them on the parent class for both. I missed that in the last pr, sorry.

You'll have to rename the file to `subnet_test.rb`, I don't think `/test/unit/subnet.rb` is actually being run now.